unstable/wlr-surface-layers-unstable-v1.xml

```diff
@@ -0,0 +1,231 @@
+    FITNESS, IN NO EVENT SHALL THE COPYRIGHT HOLDERS BE LIABLE FOR ANY
+    SPECIAL, INDIRECT OR CONSEQUENTIAL DAMAGES OR ANY DAMAGES
+    WHATSOEVER RESULTING FROM LOSS OF USE, DATA OR PROFITS, WHETHER IN
+    AN ACTION OF CONTRACT, NEGLIGENCE OR OTHER TORTIOUS ACTION,
+    ARISING OUT OF OR IN CONNECTION WITH THE USE OR PERFORMANCE OF
+    THIS SOFTWARE.
+  </copyright>
+
+  <interface name="zwlr_layer_surface_shell_v1" version="1">
+    <description summary="create surfaces that are layers of the desktop">
+      Clients can use this interface to assign the surface_layer role to
+      wl_surfaces. Such surfaces are assigned to a "layer" of the output and
+      rendered with a defined z-depth respective to each other. They may also be
+      anchored to the edges and corners of a screen and specify input handling
+      semantics. This interface should be suitable for the implementation of
+      many desktop shell components, and a broad number of other applications
+      that interact with the desktop.
+    </description>
+
+    <request name="get_layer_surface">
+      <description summary="create a layer_surface from a surface">
+        Create a layer surface for an existing surface. This assigns the role of
+        layer_surface, or raises a protocol error if another role is already
+        assigned.
+
+        Creating a layer surface from a wl_surface which has a buffer attached
+        or committed is a client error, and any attempts by a client to attach
+        or manipulate a buffer prior to the first layer_surface.configure call
+        must also be treated as errors.
+
+        Clients can specify a namespace that defines the purpose of the layer
+        surface.
+      </description>
+      <arg name="id" type="new_id" interface="zwlr_layer_surface_v1"/>
+      <arg name="surface" type="object" interface="wl_surface"/>
+      <arg name="output" type="object" interface="wl_output"/>
+      <arg name="layer" type="uint" enum="layer" summary="layer to add this surface to"/>
+      <arg name="namespace" type="string" summary="namespace for the layer surface"/>
+    </request>
+
+    <enum name="error">
+      <entry name="role" value="0" summary="wl_surface has another role"/>
+      <entry name="invalid_layer" value="1" summary="layer value is invalid"/>
+      <entry name="already_constructed" value="2" summary="wl_surface has a buffer attached or committed"/>
+    </enum>
+
+    <enum name="layer">
+      <description summary="available layers for surfaces">
+        These values indicate which layers a surface can be rendered in. They
+        are ordered by z depth, bottom-most first. Traditional shell surfaces
+        will typically be rendered between the bottom and top layers.
+        Fullscreen shell surfaces are typically rendered at the top layer.
+        Multiple surfaces can share a single layer, and ordering within a
+        single layer is undefined.
```

Comment:
I don't understand the difference between background/bottom and top/overlay. If I'm making a background or overlay application, it's easy to make the right choice, but what about other application types like bars and notifications?

What is the use case for picking "bottom" instead of "background"?

What is the use case for picking "top" instead of "overlay"?

Reply:
background: wallpaper

bottom: panel (windows can be dragged over it)

top: notifications

overlay: lockscreen

Reply:
An on screen keyboard would need to go atop of the lock screen (since it might be needed to enter the password) 
Is this worth special casing?

Reply:
This is fixed by the namespace: both surfaces will be in the overlay layer, one will have the `lockscreen` namespace, the other the `osk` namespace. The compositor will be able to put the OSK on top of the lockscreen.

We're still exploring other options to solve this issue though. Let us know if you have ideas!

Reply:
bottom: panel (windows can be dragged over it)
In some shells (Gnome comes to mind) panels are rendered on top of windows. Would it be appropriate for panels to request top in that case? Might it be a good idea to have something in between, so that shell components can let the shell decide?

Reply:
Yes, panels that render on top of other windows should choose the top layer. I don't think we need something in-between.